### PR TITLE
Synth redo

### PR DESCRIPTION
**Pull request by DJ-2805**
The new synth file is cleaned up and documented. I'm still testing it, and need to fix the scaling of the y-axis of the bottom subplot along with add the actual parabola function and some cleaning up for the plots.

### CODE REVIEW

**Comment by spencerahill** [comment]
Sounds good.  Just ping me when you're ready for a review.

**Comment by DJ-2805** [comment]
I believe my code is ready to be reviewed. The main two files are synth.py and paraboloid.py. Let me know what you think!

**Comment by spencerahill** [comment]
Great, looking forward to going through this.  Hopefully sometime this week.

For now, obviously we don't have any formal tests yet, but have you done any informal "regression" tests e.g. by comparing digipyro output applied to the same movie with and without your code changes?  Is the output unchanged, or if you intended for it to change, has it changed in the way you meant without any unintended side effects?

I'll wait for you to confirm that before going through it.  Thanks!

**Comment by DJ-2805** [comment]
Well, I've mainly updated Synth, and I can't compare that to the old Synth, because it was producing incorrect experiments (There seemed to be a hidden negative sign somewhere in the old code). I've been able to compare my output to tests from the textbook test cases (Cushman-Roisin Ch. 2), and those match up correctly. The DigiPyRo script is the next one I plan to work on.

**Comment by spencerahill** [comment]
>I've been able to compare my output to tests from the textbook test cases (Cushman-Roisin Ch. 2), and those match up correctly.

OK, that's fine for the time being.

>The DigiPyRo script is the next one I plan to work on.

That's fine.  Please do that as a separate branch / PR and don't add any further commits to this one until I've managed to review it.

Why I'm being a stickler on that: to avoid "creep", where a PR starts out trying to do one thing but then keeps growing larger and larger in scope.  That makes reviewing much harder and increases the potential of accidentally introducing bugs or other problems that you or the reviewers don't catch, lost in the large diff.

I should be able to review this tomorrow, though today not likely.

File: app/mods/synth.py [hunk 1]
```diff
@@ -1,420 +1,163 @@
-import argparse as ap
-import numpy as np
-import cv2
-import matplotlib
-from matplotlib import pyplot as plt
-from scipy.optimize import leastsq
-
-matplotlib.use("Agg")
-
+# libraries
```
Comment (spencerahill) [inline app/mods/synth.py:1]: I appreciate your intention here of thoroughly commenting the code, but for simple import statements like this they are unnecessary.  The statements are all clear enough.  Comments are only required for lines that aren't easily understandable on their own.

File: app/mods/synth.py [hunk 2]
```diff
@@ -1,420 +1,163 @@
...
+import argparse as ap                       # command line lib
```
Comment (spencerahill) [inline app/mods/synth.py:2]: Standard style is to have imports bunched into three sections, each separated by a blank line:

1. Imports from the python standard library (argparse in this case, which separately there's no need to abbreviate here to ap)
2. Imports from external packages (numpy and matplotlib in this case)
3. Imports from within this package (paraboloid in this case).

File: app/mods/synth.py [hunk 3]
```diff
@@ -1,420 +1,163 @@
...
+import paraboloid as pbd                    # paraboloid lib, paraboloid.py
+import numpy as np                          # numerical lib
+import matplotlib.pyplot as plt             # plotting lib
+import matplotlib.animation as animate      # animation lib
+
+# sets plots to a darker color scheme
```
Comment (spencerahill) [inline app/mods/synth.py:8]: Ditto comment above re: excess commenting

File: requirements.txt [hunk 4]
```diff
@@ -2,3 +2,4 @@ numpy
 scipy
```
Comment (spencerahill) [inline requirements.txt:2]: FYI as part of the package overhaul (which I'll do after we merge this PR) this requirements.txt file will get superseded by a `setup.py` file that includes all of the requirements.

File: app/mods/paraboloid.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
+import numpy as np
+
+g = 9.817
+
+def position(t, omega, u0, v0, x0):
+    """Finds position of puck on paraboloid. The equations come from
+    Cushman-Roisin Chapter 2, in particular, 2.17a, and 2.17b.
+
```
[the remark this answers is not in the document]
Comment (DJ-2805) [reply app/mods/paraboloid.py:8]: Great! I was following the standard outlined in PEP, but I can make the adjustments for this one.

**Comment by DJ-2805** [comment]
Okay, I made the updated changes for the documentation. Let me know what you think!

**Comment by spencerahill** [comment]
Great, thanks DJ.  Will try to get to this in the coming days.

**Comment by DJ-2805** [comment]
Yes, Jon and I have tested it a couple times and so far has passed our couple of tests, so I would say it's ready to merge up. Perhaps also merging it up to dev and closing this branch, since now it will be smaller changes on it now.

**Comment by spencerahill** [comment]
Thanks again @DJ-2805.  I didn't notice until your last comment that this was a PR not into master but its own branch.  Moving forward PRs can be straight into the master branch.  I'll create a "stable" branch for formal releases once we get to that point.

**Pull request merged**